### PR TITLE
[github] Use Ubuntu 24.04 images

### PR DESCRIPTION
**Pull request by Ferenc-**

### CODE REVIEW

File: .github/workflows/linux-check.yaml [hunk 1]
```diff
@@ -95,11 +95,11 @@ jobs:
 
   linux-matrix:
     name: Linux builds and tests
-    runs-on: ubuntu-22.04
+    runs-on: ubuntu-24.04
     strategy:
       fail-fast: false
       matrix:
-        compiler: [{ CXX: g++-12, CC: gcc-12 }, { CXX: clang++-14, CC: clang-14 }]
+        compiler: [{ CXX: g++-12, CC: gcc-12 }, { CXX: clang++-16, CC: clang-16 }]
```
Comment (biodranik) [inline .github/workflows/linux-check.yaml:102]:
```suggestion
        compiler: [{ CXX: g++-14, CC: gcc-14 }, { CXX: clang++-16, CC: clang-16 }]
```

**Review by biodranik — APPROVED** [review]
Can we also upgrade gcc?

File: .github/workflows/linux-check.yaml (hunk 1 above)
Comment (Ferenc-) [reply .github/workflows/linux-check.yaml]: Then I have also increased clang to the highest version.

**Review by biodranik — APPROVED** [review]
Cool, thanks! 
1. Let's check logs output and create necessary issues to fix important warnings.
2. Which other helpful warnings can be enabled?

**Pull request merged**